### PR TITLE
corrected std function

### PR DESCRIPTION
**Pull request by coquelin77**
## Description
see title

Issue/s resolved: #766 

## Changes proposed:
- corrected bug

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Due Diligence

- [ ] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [ ] Documentation updated (if needed)
- [ ] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/statistics.py [hunk 1]
```diff
@@ -1092,7 +1092,9 @@ def __moment_w_axis(
         output_shape = output_shape if output_shape else (1,)
 
         if x.split is None:  # x is *not* distributed -> no need to distribute
-            return factories.array(function(x.larray, **kwargs), dtype=x.dtype, device=x.device)
+            ret = factories.array(function(x.larray, **kwargs), dtype=x.dtype, device=x.device)
+            print("here", type(ret))
```
Comment (ClaudiaComito) [inline heat/core/statistics.py:1096]: leftover debugging?

File: heat/core/statistics.py [hunk 2]
```diff
@@ -1535,10 +1537,24 @@ def std(
     >>> ht.std(a, 1)
     DNDarray([1.2961, 0.3362, 1.0739, 0.9820], dtype=ht.float32, device=cpu:0, split=None)
     """
-    if not axis:
-        return np.sqrt(var(x, axis, ddof, **kwargs))
+    if not isinstance(ddof, int):
+        raise TypeError(f"ddof must be integer, is {type(ddof)}")
+    elif ddof > 1:
+        raise NotImplementedError("Not implemented for ddof > 1.")
```
Comment (ClaudiaComito) [inline heat/core/statistics.py:1543]: this is "only" not implemented for the distributed or GPU case, correct?

File: heat/core/statistics.py [hunk 3]
```diff
@@ -1535,10 +1537,24 @@ def std(
...
+        raise ValueError(f"Expected ddof=0 or ddof=1, got {ddof}")
     else:
-        return exponential.sqrt(var(x, axis, ddof, **kwargs), out=None)
+        if kwargs.get("bessel"):
+            unbiased = kwargs.get("bessel")
+        else:
+            unbiased = bool(ddof)
+        ddof = 1 if unbiased else ddof
+    if not x.is_distributed() and str(x.device)[:3] == "cpu":
+        loc = np.std(x._DNDarray__array.numpy(), axis=axis, ddof=ddof)
```
Comment (ClaudiaComito) [inline heat/core/statistics.py:1553]: here `ddof > 1` would be ok

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Thanks @coquelin77, just a few comments:

File: heat/core/statistics.py (hunk 2 above)
Comment (coquelin77) [reply heat/core/statistics.py:1543]: i think it wasnt allowed by torch when i wrote that. but i guess it is now. ill take that out. worst case, we put it back in later

File: heat/core/statistics.py (hunk 3 above)
Comment (coquelin77) [reply heat/core/statistics.py:1553]: oh this was because numpy's std function is/was faster than torch's

File: heat/core/statistics.py [hunk 4]
```diff
@@ -1535,10 +1537,24 @@ def std(
     >>> ht.std(a, 1)
     DNDarray([1.2961, 0.3362, 1.0739, 0.9820], dtype=ht.float32, device=cpu:0, split=None)
     """
-    if not axis:
-        return np.sqrt(var(x, axis, ddof, **kwargs))
+    if not isinstance(ddof, int):
+        raise TypeError(f"ddof must be integer, is {type(ddof)}")
```
Comment (ClaudiaComito) [inline heat/core/statistics.py:1539]: This one line is not covered by tests.

**Pull request merged**